streams/src/test/java/org/apache/kafka/streams/state/internals/CachingKeyValueStoreTest.java

```diff
@@ -122,38 +123,34 @@ public void shouldAvoidFlushingDeletionsWithoutDirtyKeys() {
     }
 
     @Test
-    public void shouldCloseWrappedStoreAfterErrorDuringCacheFlush() {
+    public void shouldCloseWrappedStoreAndCacheAfterErrorDuringCacheFlush() {
         setUpCloseTests();
+        EasyMock.reset(cache);
         cache.flush(CACHE_NAMESPACE);
-        EasyMock.expectLastCall().andThrow(new NullPointerException("Simulating an error on flush"));
+        EasyMock.expectLastCall().andThrow(new RuntimeException("Simulating an error on flush"));
         EasyMock.replay(cache);
         EasyMock.reset(underlyingStore);
         underlyingStore.close();
         EasyMock.replay(underlyingStore);
 
-        try {
-            store.close();
-        } catch (final RuntimeException exception) {
-            EasyMock.verify(underlyingStore);
-        }
+        assertThrows(RuntimeException.class, store::close);
+        EasyMock.verify(cache, underlyingStore);
```

Comment:
I added the verification for the call to `close()` of underlying store and cache.